src/api/v4/leaflet-layer.js

```diff
@@ -0,0 +1,110 @@
+var L = require('leaflet');
+var Layer = require('./layer');
+var LeafletCartoLayerGroupView = require('../../geo/leaflet/leaflet-cartodb-layer-group-view');
+
+/**
+ * This object is a custom Leaflet layer to enable feature interactivity
+ * using an internal LeafletCartoLayerGroupView instance.
+ * 
+ * There are some overwritten functions:
+ * - getAttribution: returns always a custom OpenStreetMap / Carto attribution message
+ * - addTo: when the layer is added to a map it also creates a LeafletCartoLayerGroupView
+ *          object called `_internalView` in order to enable the feature events
+ * - removeFrom: when the layer is removed from a map it also removes the feature events
+ *               listeners, triggers a 'remove' event and removes the `_internalView`
+ *
+ * NOTE: It also contains the feature events handlers. That's why it requires the carto layers array.
+ */
+
+var LeafletLayer = L.TileLayer.extend({
```

Comment:
Should we document this and link it from `client.getLeafletLayer`? I would make it clear that this is a L.TilerLayer (also, it would be interesting to hide `setUrl` from final users).

Reply:
The `client.getLeafletLayer` function documentation contains the  @ returns L.TileLayer with a link. I think it is enough information for the user, because this is not a wrapper.

We can overwrite the `setUrl, createTile` with empty content OR use a `L.GridLayer` instead that does not contains these functions.